In [3]:
import pandas as pd

In [4]:
from ydata_profiling import ProfileReport

In [5]:
df=pd.read_csv('diabetes.csv')

In [6]:
df.head(10)

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   
5            5      116             74              0        0  25.6   
6            3       78             50             32       88  31.0   
7           10      115              0              0        0  35.3   
8            2      197             70             45      543  30.5   
9            8      125             96              0        0   0.0   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1  
5                     0.201   30        0  
6                     0.248   26        1  
7                     0.134   29        0  
8                     0.158   53        1  
9                     0.232   54        1

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               768 non-null    int64  
 1   Glucose                   768 non-null    int64  
 2   BloodPressure             768 non-null    int64  
 3   SkinThickness             768 non-null    int64  
 4   Insulin                   768 non-null    int64  
 5   BMI                       768 non-null    float64
 6   DiabetesPedigreeFunction  768 non-null    float64
 7   Age                       768 non-null    int64  
 8   Outcome                   768 non-null    int64  
dtypes: float64(2), int64(7)
memory usage: 54.1 KB


In [8]:
df.isna().sum()

Pregnancies                 0
Glucose                     0
BloodPressure               0
SkinThickness               0
Insulin                     0
BMI                         0
DiabetesPedigreeFunction    0
Age                         0
Outcome                     0
dtype: int64

In [9]:
zero_count=(df==0).sum()
print(zero_count)

Pregnancies                 111
Glucose                       5
BloodPressure                35
SkinThickness               227
Insulin                     374
BMI                          11
DiabetesPedigreeFunction      0
Age                           0
Outcome                     500
dtype: int64


In [10]:
df.describe()

Pregnancies     Glucose  BloodPressure  SkinThickness     Insulin  \
count   768.000000  768.000000     768.000000     768.000000  768.000000   
mean      3.845052  120.894531      69.105469      20.536458   79.799479   
std       3.369578   31.972618      19.355807      15.952218  115.244002   
min       0.000000    0.000000       0.000000       0.000000    0.000000   
25%       1.000000   99.000000      62.000000       0.000000    0.000000   
50%       3.000000  117.000000      72.000000      23.000000   30.500000   
75%       6.000000  140.250000      80.000000      32.000000  127.250000   
max      17.000000  199.000000     122.000000      99.000000  846.000000   

              BMI  DiabetesPedigreeFunction         Age     Outcome  
count  768.000000                768.000000  768.000000  768.000000  
mean    31.992578                  0.471876   33.240885    0.348958  
std      7.884160                  0.331329   11.760232    0.476951  
min      0.000000                  0.078000   21.000000    0.000000  
25%     27.300000                  0.243750   24.000000    0.000000  
50%     32.000000                  0.372500   29.000000    0.000000  
75%     36.600000                  0.626250   41.000000    1.000000  
max     67.100000                  2.420000   81.000000    1.000000

In [11]:
profile=ProfileReport(df,title='Rappot diabetes')
profile.to_file('Rappot diabetes')

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

c:\Users\nabyp\OneDrive\Bureau\Analyse de données\PYTHON DATA\sklearn-env\Lib\site-packages\ydata_profiling\profile_report.py:379: UserWarning: Extension  not supported. For now we assume .html was intended. To remove this warning, please use .html or .json.
  warnings.warn(


Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

#       OBSERVATIONS SUR LES DONNEES

#  APERCU GENERAL 

A l'issue de l'exploration des données nous pouvons constater que notre dataset contient 768 enregistrements(lignes) et compte 9 variables (colonnes).La majorité des variables sont numeriques :
1 catégoriques et  8 numériques.
On remarque cependant que notre dataset est propre c'est à dire qu'il n'y a pas de lignes en doubles encore moins de données manquantes.
Aussi on se rend compte que notre dataset  ne consomme pas trop en memoire , il occupe 54.1KB ,soit 72.2B par enregistrement.

#   Observations sur les variables 

#   -PREGNANCIES

Type de donnée : Nombre réel (ℝ)

-Corrélation élevée : Indique que la variable "Pregnancies" pourrait avoir une relation significative avec une ou plusieurs autres variables du dataset.

-Valeurs nulles : 0 (0.0%) Il n'y a pas de valeurs manquantes pour cette variable, ce qui est idéal pour les analyses.

-Moyenne : 3.845 .En moyenne, les participantes de l'étude ont eu environ 3,8 grossesses.

Minimum : 0 Certaines participantes n'ont eu aucune grossesse, ce qui est indiqué par les valeurs minimales de zéro.

Maximum : 17 La valeur maximale de grossesses est de 17, ce qui peut représenter des cas extrêmes ou particuliers.

Zeros : 111 (14.5%). Un pourcentage significatif de 14.5% des participants n'ont pas eu de grossesse. Cela est important à noter car cela peut influencer les analyses statistiques.

Valeurs négatives : Aucune valeur négative, ce qui est correct pour le nombre de grossesses.

Conclusion :

La variable "Pregnancies" présente des caractéristiques intéressantes à prendre en compte :

Distribution des valeurs : Une distribution variée avec des valeurs allant de 0 à 17 grossesses.

Corrélation élevée : Indique des relations significatives avec d'autres variables, ce qui peut être exploré davantage.

#   -Glucose

-Type de donnée : Nombre réel (ℝ)

-Valeurs distinctes : 136 (17.7%)
Il y a 136 valeurs distinctes pour le glucose dans notre dataset, ce qui montre une bonne variabilité des données.

-Moyenne : La moyenne du niveau de glucose est d'environ 120.89, ce qui peut être utile pour comprendre la distribution générale des niveaux de glucose dans notre dataset  et ainsi déterminer les niveaux de glucose normaux.

-Minimum :La présence de valeurs égales à zéro 5 (0.7%) ,pourrait être problématique, car les niveaux de glucose de zéro ne sont pas réalistes et pourraient indiquer des valeurs manquantes ou des erreurs de saisie de données.

-Maximum : Le niveau de glucose le plus élevé est de 199, ce qui peut indiquer des cas potentiels de diabète ou d'hyperglycémie.

Conclusion:

La variable glucose dans est majoritairement propre et bien représentée. Les points principaux à noter sont :

Valeurs zéro : Vérifiez les valeurs zéro, car elles ne sont pas réalistes pour les niveaux de glucose et pourraient indiquer des données manquantes ou des erreurs.

Distribution générale : La variabilité et la moyenne des valeurs de glucose semblent raisonnables et peuvent fournir une bonne base pour des analyses supplémentaires.

#   -BMI (IMC)

BMI (Body Mass Index)

Moyenne : 31.99

Minimum : 0

Maximum : 67.1

Zeros : 1.4% (11 occurrences)

La moyenne de BMI à 31.99 indique que de nombreux individus de l'échantillon sont en surpoids ou obèses, selon les classifications standards.

La présence de valeurs zéro (1.4%) est faible mais notable, et pourrait également représenter des erreurs ou des données manquantes.

La distribution des valeurs BMI est large, avec un minimum de 0 et un maximum de 67.1, couvrant un large éventail d'IMC.

#   -BloodPressure


Type de donnée : Nombre réel (ℝ)

Distinct : 47 (6.1%) ,

Valeurs manquantes : 0 (0.0%)

Valeurs infinies : 0 (0.0%)

Moyenne : 69.11

Minimum : 0

Maximum : 122

Zeros : 35 (4.6%)

La moyenne est egale à 69.11 ce qui indique que la moyenne de l'echatillon on une pression sanguine proche de la  normale (75-80)

La présence de zéros dans BloodPressure pourrait indiquer des valeurs manquantes ou des erreurs de saisie. La distribution est principalement correcte, mais ces zéros doivent être examinés.

#   -SkinThickness

Type de donnée : Nombre réel (ℝ)

Distinct : 51 (6.6%)

Valeurs manquantes : 0 (0.0%)

Valeurs infinies : 0 (0.0%)

Moyenne : 20.54

Minimum : 0

Maximum : 99

Zeros : 227 (29.6%)

Un pourcentage élevé de zéros dans SkinThickness suggère que cette variable pourrait avoir de nombreuses valeurs manquantes représentées par zéro. Il est crucial de déterminer comment gérer ces valeurs.

#   -Insulin


Type de donnée : Nombre réel (ℝ)

Distinct : 186 (24.2%)

Valeurs manquantes : 0 (0.0%)

Valeurs infinies : 0 (0.0%)

Moyenne : 79.80

Minimum : 0

Maximum : 846

Zeros : 374 (48.7%)

Près de la moitié des valeurs pour Insulin sont zéro, ce qui indique probablement des valeurs manquantes. Cela mérite une attention particulière pour décider comment imputer ces données.

#   -DiabetesPedigreeFunction

Type de donnée : Nombre réel (ℝ)

Distinct : 517 (67.3%)

Valeurs manquantes : 0 (0.0%)

Valeurs infinies : 0 (0.0%)

Moyenne : 0.47

Minimum : 0.078

Maximum : 2.42

Zeros : 0 (0.0%)

Aucune valeur zéro ou manquante pour DiabetesPedigreeFunction, on note une variabilité remarquable sur la distribution ce qui est idéal pour l'analyse.


#   -Age

Type de donnée : Nombre réel (ℝ)

Distinct : 52 (6.8%)

Valeurs manquantes : 0 (0.0%)

Valeurs infinies : 0 (0.0%)

Moyenne : 33.24

Minimum : 21

Maximum : 81

Zeros : 0 (0.0%)

Les valeurs d'âge sont complètes et correctes, sans valeurs manquantes ou anormales.

#   -Outcome


Type de donnée : Catégorielle

Distinct : 2 (0.3%)

Valeurs manquantes : 0 (0.0%)

La variable Outcome est binaire et complète, ce qui est idéal pour une variable cible dans un modèle de classification.



#   -Interprétation des corrélations

-Age :

Corrélation élevée avec Pregnancies (0.607) : Les femmes plus âgées ont tendance à avoir eu plus de grossesses, ce qui est une relation attendue.

Corrélation modérée avec BloodPressure (0.351) : Cela pourrait indiquer que la pression artérielle augmente avec l'âge.

-BMI :

Corrélation modérée avec SkinThickness (0.444) : Cela peut s'expliquer par le fait que les personnes avec un IMC élevé peuvent avoir une épaisseur de peau plus importante.

Corrélation modérée avec Outcome (0.317) : Une corrélation positive indique qu'un IMC plus élevé pourrait être associé à un risque accru de diabète.

-BloodPressure :

Corrélation modérée avec Age (0.351) : Comme mentionné plus haut, la pression artérielle semble augmenter avec l'âge.

Corrélation modérée avec BMI (0.293) : Cela suggère que l'IMC et la pression artérielle peuvent être liés.

-Glucose :

Corrélation élevée avec Outcome (0.487) : Les niveaux de glucose élevés sont fortement associés à un diagnostic de diabète.

Corrélation modérée avec Age (0.285) : Indique que les niveaux de glucose augmentent légèrement avec l'âge.

-Insulin :

Corrélation élevée avec SkinThickness (0.541) : Les niveaux d'insuline et l'épaisseur de la peau semblent être liés, ce qui pourrait être exploré davantage.

Corrélation modérée avec Glucose (0.213) : Une certaine relation entre les niveaux de glucose et d'insuline est observée, ce qui est attendu.

-Outcome (Résultat du test de diabète) :

Corrélation élevée avec Glucose (0.487) : Comme mentionné plus haut, les niveaux de glucose élevés sont fortement associés à un diagnostic de diabète.

Corrélation modérée avec BMI (0.317) : Indique qu'un IMC plus élevé est associé à un risque accru de diabète.

-Pregnancies :

Corrélation élevée avec Age (0.607) : Déjà discuté, plus les femmes sont âgées, plus elles ont tendance à avoir eu des grossesses.

Corrélation modérée avec Outcome (0.235) : Indique une légère relation où le nombre de grossesses pourrait influencer le risque de diabète.

Conclusion

Les corrélations entre ces variables  donnent un aperçu des relations potentielles et des associations dans notre dataset. Les corrélations fortes indiquent des relations intéressantes qui peuvent être explorées plus en détail, tandis que les corrélations faibles peuvent suggérer des relations moins significatives.

#   -Conclusion

Le  dataset est globalement bien structuré, avec certaines variables nécessitant une attention particulière en raison des valeurs zéro (pouvant indiquer des valeurs manquantes). Les variables comme Insulin, SkinThickness, et BloodPressure présentent des valeurs zéro significatives qui doivent être analysées et traitées pour garantir des analyses précises.

